driver-core/src/test/functional/com/mongodb/internal/connection/InternalStreamConnectionFunctionalSpecification.groovy

```diff
@@ -0,0 +1,111 @@
+package com.mongodb.internal.connection
+
+import com.mongodb.OperationFunctionalSpecification
+import com.mongodb.ReadPreference
+import com.mongodb.ServerAddress
+import com.mongodb.connection.ByteBufferBsonOutput
+import com.mongodb.connection.ClusterConnectionMode
+import com.mongodb.connection.ClusterId
+import com.mongodb.connection.ServerId
+import com.mongodb.connection.ServerType
+import com.mongodb.connection.ServerVersion
+import com.mongodb.connection.SocketSettings
+import com.mongodb.connection.SocketStreamFactory
+import com.mongodb.connection.SslSettings
+import com.mongodb.internal.validator.NoOpFieldNameValidator
+import org.bson.BsonDocument
+import org.bson.BsonInt32
+import org.bson.BsonString
+import org.bson.codecs.BsonDocumentCodec
+import spock.lang.IgnoreIf
+
+import java.util.concurrent.TimeUnit
+
+import static com.mongodb.ClusterFixture.serverVersionAtLeast
+
+class InternalStreamConnectionFunctionalSpecification extends OperationFunctionalSpecification {
+
+    def serverId = new ServerId(new ClusterId(), new ServerAddress('localhost', 27017))
+    def socketSettings = SocketSettings.builder().connectTimeout(1000, TimeUnit.MILLISECONDS).build()
+    def sslSettings = SslSettings.builder().build()
+    InternalStreamConnection connection = new InternalStreamConnection(serverId, new SocketStreamFactory(socketSettings, sslSettings), [],
+            new TestCommandListener(), new InternalStreamConnectionInitializer([], null, []))
+
+    def fieldNameValidator = new NoOpFieldNameValidator()
+    def sessionContext = NoOpSessionContext.INSTANCE
+
+    def findCommand = new BsonDocument('find', new BsonString(getCollectionName())).append('batchSize', new BsonInt32(1))
+    def findMessage = new CommandMessage(namespace, findCommand, fieldNameValidator, ReadPreference.primary(),
+            MessageSettings.builder().serverVersion(new ServerVersion([4, 1, 3])).serverType(ServerType.STANDALONE).build(),
+            true, null, null, ClusterConnectionMode.SINGLE).setExhaust(false)
+
+    @IgnoreIf({ !serverVersionAtLeast([4, 1, 3]) })
+    def 'should respond with moreToCome flagbit set'() {
```

Comment:
Is this test redundant to the one after?  The assertion it makes is unsatisfying, as it doesn't prove anything about how the class actually responds to the flag bits being equal to 2.

Reply:
Yeah, it's a bit redundant so i've removed it!